### PR TITLE
Adding ability to create a new 'folder'

### PR DESCRIPTION
**Pull request by kiiadi**

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/NewFolderAction.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    val bucket: S3VirtualBucket
+) : ActionButtonWrapper(message("s3.new.folder"), null, null) {
+
+    override fun isEnabled(): Boolean = treeTable.selectedRows.size <= 1
+
+    override fun doActionPerformed(e: AnActionEvent) {
+
+        val node = treeTable.selectedRows.firstOrNull()?.let { treeTable.getNodeForRow(it) } ?: treeTable.getRootNode()
+
+        Messages.showInputDialog(e.project, message("s3.new.folder.name"), message("s3.new.folder"), null)?.let { key ->
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/NewFolderAction.kt]: Do we want to block `/` in the entered name?

Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/NewFolderAction.kt]: you mean just plain ol "/" or anything with a "/" in it?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/NewFolderAction.kt]: Both.

Should we allow foo/bar/file?
Or /foo/ seems odd as well

Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/NewFolderAction.kt]: I think we should leave the validation to S3, I wonder what /folder would do

Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/NewFolderAction.kt]: /folder "works" it just creates a blank node with a sub-folder of "folder"

**Pull request merged**